supercollider/server.py

```diff
@@ -13,188 +16,131 @@ def __init__(self, hostname="127.0.0.1", port=57110):
+    def dummy_handler(self, address, *args):
+        pass
+
+    def simple_handler(self, address, *args):
+        return args
+
+    def param_handler(self, address, *args):
+        return args[2]
+
+    def buf_handler(self, address, *args):
+        return args[3:]
+
+    def status_handler(self, address, *args):
+        status_dict = {
                 "num_ugens": args[1],
                 "num_synths": args[2],
                 "num_groups": args[3],
                 "num_synthdefs": args[4],
                 "cpu_average": args[5],
                 "cpu_peak": args[6],
                 "sample_rate_nominal": args[7],
-                "sample_rate_actual": args[8],
-            }
-            if callback:
-                callback(args_dict)
-            return args_dict
+                "sample_rate_actual": args[8]}
+            
+        return status_dict
+
+    def version_handler(self, *args):
+        version_dict = {
+                "program_name": args[1],
+                "version_major": args[2],
+                "version_minor": args[3],
+                "version_patch": args[4],
+                "git_branch": args[5],
+                "commit_hash": args[6]}
+            
+        return version_dict
+    
+    # ------------------- Client Messages ------------------- #
+    def _send_msg(self, address, *args):
+        self.sc_client.send_message(address, [*args])
 
+    def sync(self, ping_id = 1):
+        self._send_msg("/sync", ping_id)
+        return self._await_response("/synced", None, self.simple_handler)
+
+    def query_tree(self, group=None):
+        self._send_msg("/g_queryTree", group.id if group else 0, 0)
+        return self._await_response("/g_queryTree.reply", callback=self.simple_handler)
+    
+    def get_status(self):
         self._send_msg("/status")
-        if blocking:
-            return self._await_response("/status.reply", None, _handler)
-        elif callback:
-            self._add_handler("/status.reply", None, _handler)
 
-    def get_version(self, callback=None, blocking=True):
-        """
-        Returns the current Server version.
-
-        Example:
-            >>> server.version
-            {
-                'program_name': "scsynth",
-                'version_major': 3,
-                'version_minor': 10,
-                'version_patch': ".3",
-                'git_branch': "HEAD",
-                'commit_hash': "67a1eb18"
-            }
-        """
-        def _handler(args):
-            args_dict = {
-                "program_name": args[0],
-                "version_major": args[1],
-                "version_minor": args[2],
-                "version_patch": args[3],
-                "git_branch": args[4],
-                "commit_hash": args[5]
-            }
-            if callback:
-                callback(args_dict)
-            return args_dict
+        return self._await_response("/status.reply", None, self.status_handler)
 
+    def get_version(self):
```

Comment:
Please restore the pydocs for `get_version` and `get_status` - they seem to have been lost in this fork

Reply:
Solved in [9bd584f](https://github.com/ideoforms/python-supercollider/pull/12/commits/9bd584f53a5ec7a0eed072ca67869c326910dcdc)